gammapy/estimators/utils.py

```diff
@@ -461,3 +462,127 @@ def compute_lightcurve_doublingtime(lightcurve, flux_quantity="flux"):
     )
 
     return table
+
+
+def get_edges_fixed_bins(fluxpoint, group_size, axis_name="energy"):
+    """Rebin the flux point to combine value adjacent bins
+
+    Parameters
+    ----------
+    fluxpoint : `FluxPoints`
+        The flux point to rebin
+    group_size : int
+        Number of bins to combine
+
+    Returns
+    -------
+    edges_min : `Quantity` or `Time`
+    edges_max : `Quantity` or `Time`
+        Low and High edges of the new axis
+    """
+
+    ax = fluxpoint.geom.axes[axis_name]
+    nbin = ax.nbin
+    if not isinstance(group_size, int):
+        raise ValueError("Only integer number of bins can be combined")
+    idx = np.arange(0, nbin, group_size)
+    if idx[-1] < nbin:
+        idx = np.append(idx, nbin)
+    edges_min = ax.edges_min[idx[:-1]]
+    edges_max = ax.edges_max[idx[1:] - 1]
+    return edges_min, edges_max
+
+
+def get_edges_min_sqrt_ts(fluxpoint, sqrt_ts_threshold, axis_name="energy"):
+    """Rebin the flux point to combine adjacent bins
+       till a minimum significance is obtained
+
+    Parameters
+    ----------
+    fluxpoint : `FluxPoints`
+            The flux point to rebin
+    sqrt_ts_threshold : float
+        The minimum significance desired
+
+    Returns
+    -------
+    edges_min : `Quantity` or `Time`
+    edges_max : `Quantity` or `Time`
+        Low and High edges of the new axis
+    """
+    ax = fluxpoint.geom.axes[axis_name]
+    nbin = ax.nbin
+
+    e_min, e_max = ax.edges_min[0], ax.edges_max[0]
+    edges_min = np.zeros(nbin) * e_min.unit
+    edges_max = np.zeros(nbin) * e_max.unit
+    i, i1 = 0, 0
+    while e_max < ax.edges_max[-1]:
+        ts = fluxpoint.ts.data[i]
+        e_min = ax.edges_min[i]
+        while ts < sqrt_ts_threshold**2 and i < ax.nbin - 1:
+            i = i + 1
+            ts = ts + fluxpoint.ts.data[i]
+        e_max = ax.edges_max[i]
+        i = i + 1
+        edges_min[i1] = e_min
+        edges_max[i1] = e_max
+        i1 = i1 + 1
+    edges_max = edges_max[:i1]
+    edges_min = edges_min[:i1]
+
+    return edges_min, edges_max
+
+
+RESAMPLE_METHODS = {
+    "fixed-bins": get_edges_fixed_bins,
+    "min-sqrt-ts": get_edges_min_sqrt_ts,
+}
+
+
+def get_rebinned_axis(fluxpoint, axis_name="energy", **kwargs):
+    """Get the rebinned axis for resampling
+     the flux point object along the mentioned axis.
+
+
+    Parameters
+    ----------
+    kwargs : Dict
```

Comment:
Why not explictly describe the arguments in the `Parameters`? 
`kwargs` is not optional here since you need to pass the `method`. You should rather list it as a required argument.
I would also list the other arguments as parameters.